src/main/java/core/basesyntax/shop/strategy/FileReadStrategy.java

```diff
@@ -0,0 +1,16 @@
+package core.basesyntax.shop.strategy;
+
+import core.basesyntax.shop.service.ReadFromFile;
+import core.basesyntax.shop.service.impl.ReadFromCsvFile;
+
+public class FileReadStrategy {
```

Comment:
You may write all this checks in your ReadFromCsvFile class if you wish to. So seems you just don't need this class.

Reply:
It's needed for further expanding with other filetypes

Reply:
It's violation of Open/Close principle.

Reply:
No. It's just a checking of file extension to choose appropriate strategy

Reply:
and i have separate handlers. in this strategy class it is decided which one to chose